scripts/valhalla_build_elevation

```diff
@@ -1,81 +1,224 @@
+    default=0,
+)
+
+Tile = namedtuple("Tile", ["name", "dir"])
+
+
+def get_tile_info(
+    x: int,
+    y: int,
+) -> Tile:
+    hemisphere = "S" if y < 0 else "N"
+    dir_name = "%s%02d" % (hemisphere, abs(y))
+    tile_name = "%s%02d%s%03d.hgt" % (
+        hemisphere,
+        abs(y),
+        "W" if x < 0 else "E",
+        abs(x),
+    )
+
+    return Tile(tile_name, dir_name)
+
+
+def get_tiles_with_shapely(input_dir: Path) -> List[Tile]:
+    try:
+        from shapely.geometry import Polygon, box
+    except ImportError:
+        LOGGER.critical(
+            "Could not import shapely. Please install shapely or use the --bbox option instead."
+        )
+        sys.exit(1)
+
+    tile_infos = []
+    for coords in get_outer_rings(input_dir):
+        poly = Polygon(coords)
+        for rect in grid_from_bounds(poly.bounds):
+            if poly.intersects(box(*rect)):
+                s, w, *_ = rect
+                tile_infos.append(get_tile_info(s, w))
+
+    return tile_infos
+
+
+def get_outer_rings(input_dir: Path) -> List:
+    for file in input_dir.glob("*.geojson"):
+        with open(file) as f:
+            geojson = json.load(f)
+        for feature in geojson["features"]:
+            if feature["geometry"]["type"] == "Polygon":
+                yield feature["geometry"]["coordinates"][0]  # outer ring only
+            if feature["geometry"]["type"] == "MultiPolygon":
+                for single_polygon in feature["geometry"]["coordinates"]:
+                    yield single_polygon[0]  # outer ring only
+
+
+def grid_from_bounds(bounds: List[float]) -> List[List[int]]:
+    """Creates a regular grid of size 1x1 within specified bounds"""
+    # loop through x and y range and create the grid
+    min_x, min_y = [floor(x) for x in bounds[:2]]
+    max_x, max_y = [ceil(x) for x in bounds[-2:]]
+    for x in range(min_x, max_x):
+        for y in range(min_y, max_y):
+            yield [x, y, x + 1, y + 1]
+
+
+def get_tiles_with_bbox(bbox_str: str) -> List[Tile]:
+    try:
+        bbox = [float(x) for x in bbox_str.split(",")]
+    except ValueError:
+        LOGGER.critical(f"BBOX {bbox_str} is not a comma-separated string of coordinates.")
+        sys.exit(1)
+
+    if not bbox_is_valid(bbox):
+        LOGGER.critical(f"Bbox invalid: {bbox}")
+        sys.exit(1)
+
+    LOGGER.debug(f"Received valid bbox: {bbox}")
+
+    tile_infos = []
+    for grid in grid_from_bounds(bbox):
+        s, w = grid[:2]
+        tile_infos.append(get_tile_info(s, w))
+
+    return tile_infos
+
+
+def bbox_is_valid(bbox):
+    min_x, min_y, max_x, max_y = bbox
+    if (
+        min_x > max_x
+        or min_x < -180
+        or max_x > 180
+        or min_y > max_y
+        or min_y < -90
+        or max_y > 90
+    ):
+        return False
+
+    return True
+
+
+def download(tile: Tile, output_dir, decompress):
+    dest_directory = Path(output_dir, tile.dir)
+    Path.mkdir(dest_directory, parents=True, exist_ok=True)
```

Comment:
Just realized we either need to lock the mkdir call here or create all parent directories first in the main thread @chrstnbwnkl . Preferably use a lock, somewhat nicer to also test eventually IMO

Reply:
I just added a lock, but thinking about it further, I think we might not need it since `exist_ok=True` catches any `OSError` that occur when creating a directory that already exists (see source [here](https://github.com/python/cpython/blob/80d9ff16483b6c1898bcdcc811b5450b57a5e573/Lib/pathlib.py#L1168). If we want to further make sure this `OSError` does not occur for other reasons, we could further inspect it as described [here](https://stackoverflow.com/a/273227/10955832). What do you think?

Reply:
> I think we might not need it since exist_ok=True catches any OSError that occurs when creating a directory that already exists

yea you're right! then we revert that change. fwiw, the lock wouldn't have an effect here since it's created for each thread separately. it's important that the lock is shared between the threads. smth like this (answer is for processes, but might hold for threads): https://stackoverflow.com/a/25558333/2582935